### ISSUE TITLE
Better documentation on private registry usage

### ISSUE THREAD

**Issue by kingdonb** [weaveworks/wksctl#140]
There is an environments/ in wksctl repo which explains how to start a registry and copy the required images for wks into it, but it is not clear from README.md or script source what else if anything is needed to make the K8s API able to pull from that registry, if special handling is needed for pods generally to start from insecure registries (I know this is a configuration option in Docker `insecure-registries`, maybe it just works by default in Kubernetes... not sure?), guidance about how to install WKS from your image source, or other basic prerequisites.

I have some private builds that I would like to push to some private registry and install to WKS, is exactly what specific use case has driven me to report this as an issue... the source code is private and unpublished, so the images should be too.  Maybe with something like a private CI / docker builder can be incorporated as a separate use case, private builds on private registry? (I was planning on building them locally and pushing them to the private registry, I am using WKS on footloose.)

I know that some flags were added to pass in the controller image, also saw dependencies.toml in the latest merges for alpha.  Other than perhaps in wks-controller.yaml, is there more configuration needed to be performed before WKSctl can self-bootstrap from a private registry?

**Comment by kingdonb** [comment]
I found the integration testing subdir where /tmp/registry is also called out, that might get me everything that I needed!

**Comment by kingdonb** [comment]
I found `examples/footloose/cluster.yaml` which gives an example of how to configure `daemon.json` for Docker's benefit, like for example setting `insecure-registries` as mentioned above, (this configuration is placed in `docker-config.yaml` from the firekube demo, 👍)

I think I have everything... hopefully these breadcrumbs will help anyone who finds this, or I'd be happy to contribute some docs or a blog post if people think this will be a common use case!

**Comment by palemtnrider** [comment]
hi @kingdonb a markdown file of what you did would be very useful.  If you could create a PR that would be appreciated!

**Comment by kingdonb** [comment]
I provided some notes in a markdown file on #144 and tried to integrate them with the wksctl docs; copy editing is welcome there.  I just downgraded my home internet pipe, and Docker Hub isn't so fast anyway, so this turned out to be really beneficial for speeding up repeatedly doing a WKS cluster startup!

As I have been able to bootstrap an effectively "air-gapped" cluster from a private, insecure registry, I'll be closing this.
